build/allow_trust_test.go

```diff
@@ -0,0 +1,112 @@
+package build
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/stellar/go-stellar-base"
+	"github.com/stellar/go-stellar-base/xdr"
+)
+
+var _ = Describe("Payment Mutators", func() {
+
+	var (
+		subject AllowTrustBuilder
+		mut     interface{}
+
+		address = "GAXEMCEXBERNSRXOEKD4JAIKVECIXQCENHEBRVSPX2TTYZPMNEDSQCNQ"
+		bad     = "foo"
+	)
+
+	JustBeforeEach(func() {
+		subject = AllowTrustBuilder{}
+		subject.Mutate(mut)
+	})
+
+	Describe("Trustor", func() {
+		Context("using a valid stellar address", func() {
+			BeforeEach(func() { mut = Trustor{address} })
+
+			It("succeeds", func() {
+				Expect(subject.Err).NotTo(HaveOccurred())
+			})
+
+			It("sets the destination to the correct xdr.AccountId", func() {
+				aid, _ := stellarbase.AddressToAccountId(address)
+				Expect(subject.AT.Trustor.MustEd25519()).To(Equal(aid.MustEd25519()))
+			})
+		})
+
+		Context("using an invalid value", func() {
+			BeforeEach(func() { mut = Trustor{bad} })
+			It("failed", func() { Expect(subject.Err).To(HaveOccurred()) })
```

Comment:
Tip:  There is a `MatchError` matcher in gomega that gives you a bit more specificity on errors that occur.  For example, you could write:

`Expect(subject.Err).To(MatchError("invalid trustor"))`

to match using a string against `subject.Err.Error()`, or you can do:

`Expect(subject.Err).To(MatchError(errors.New("some error"))` 

to match against a specific instance.

----

I'm not saying you should to change anything, just giving tips.